code/extensions/BasisFormProtectionExtension.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+class BasisFormProtectionExtension extends Extension
+{
+	private static $controllers = array();
```

Comment:
Could you please add some more comments around this class, specifically one for each method? It took me a little to notice what was actually being done here, and I feel that a few comments will help others further down the line.